File: packages/python3/model/game.py
from .player import *
from .ball import *
from .nitro_pack import *
from .robot import *


class Game:
    def __init__(self, json):
        self.current_tick = json.get("current_tick")
        self.players = list(map(Player, json.get("players")))
        self.robots = list(map(Robot, json.get("robots")))
        self.nitro_packs = list(map(NitroPack, json.get("nitro_packs")))
        self.ball = Ball(json.get("ball"))

File: packages/ruby/model/game.rb
require './model/player'
require './model/ball'
require './model/nitro_pack'
require './model/robot'

class Game
    attr_reader :current_tick
    attr_reader :players
    attr_reader :robots
    attr_reader :nitro_packs
    attr_reader :ball
    def initialize(json)
        @current_tick = json["current_tick"]
        @players = json["players"].map { |o| Player.new(o) }
        @robots = json["robots"].map { |o| Robot.new(o) }
        @nitro_packs = json["nitro_packs"].map { |o| NitroPack.new(o) }
        @ball = Ball.new(json["ball"])
    end
end